### PR TITLE
Script DSL serialization

### PR DESCRIPTION
**Pull request by joshlory**
Currently .script files can be ingested from a file into the DB, but never written from the DB into a .script file.  This change adds a serialization for the `ScriptDSL` format.

In the future this will allow editing `Script` attributes through a UI instead of hand-editing DSL text.

To get an idea of how this will affect existing .script files check out this diff: https://github.com/code-dot-org/code-dot-org/commit/1b6218283adbbdbdf72b8d6bbfd23f790dbc1bbe.

### CODE REVIEW

**Comment by joshlory** [comment]
This is an intermediate step to begin extracting `Script` attributes into dedicated edit fields (instead of the free-text DSL editor).  Once everything has a dedicated UI we should be able to change the underlying format to JSON and remove `ScriptDSL` entirely.

File: dashboard/app/dsl/script_dsl.rb [hunk 1]
```diff
@@ -121,4 +121,72 @@ def i18n_strings
...
+      jigsaw: 7,
+    }.with_indifferent_access
+    s << "id '#{legacy_script_ids[script.name]}'" if legacy_script_ids[script.name]
+
+    s << "professional_learning_course '#{script.professional_learning_course}'" if script.professional_learning_course
+    s << "peer_reviews_to_complete #{script.peer_reviews_to_complete}" if script.peer_reviews_to_complete
+
+    s << "hidden 'false'" unless script.hidden
+    s << "login_required 'true'" if script.login_required
+    s << 'hideable_stages true' if script.hideable_stages
```
Comment (wjordan) [inline dashboard/app/dsl/script_dsl.rb:144]: is this intentional for hideable_stages to be a `true` boolean argument while the others are `'true'`/`'false'` string arguments?

File: dashboard/app/models/level.rb [hunk 2]
```diff
@@ -329,7 +329,7 @@ def channel_backed?
   end
 
   def key
-    if level_num == 'custom'
+    if level_num == 'custom' || level_num.nil?
```
Comment (wjordan) [inline dashboard/app/models/level.rb:332]: is this needed to fix an unrelated bug, or related to this PR somehow?

File: dashboard/app/dsl/script_dsl.rb (hunk 1 above)
Comment (joshlory) [reply dashboard/app/dsl/script_dsl.rb:144]: You're right, the others should also be unquoted booleans.

File: dashboard/app/models/level.rb (hunk 2 above)
Comment (joshlory) [reply dashboard/app/models/level.rb:332]: Needed for this PR.  Without this change `key` is incorrect for DSL levels.  E.g.:

``` diff
-level 'VideoGamesCoordinatePlanes'
+level 'blockly:Unplugged:'

-level 'Coordinates Multi 1'
+level 'blockly:Multi:'
```

**Pull request merged**